File: MATEMÁTICA/1161/main.rb
# frozen_string_literal: true

def factorial(n, lookup)
  return 1 if n < 2

  lookup[n] = n * factorial(n - 1, lookup) unless lookup[n]
  lookup[n]
end

lookup = {}
while (line = gets)
  m, n = line.split.map(&:to_i)
  puts factorial(m, lookup) + factorial(n, lookup)
end

File: MATEMÁTICA/1161/main.py
import sys


def factorial(n, lookup):
    if n < 2:
        return 1
    if not lookup.get(n):
        lookup[n] = n * factorial(n-1, lookup)
    return lookup[n]


lookup = {}
for data in sys.stdin:
    m, n = tuple(map(int, data.split()))
    print(factorial(m, lookup) + factorial(n, lookup))
